src/System.Diagnostics.Process/src/System/Collections/Specialized/StringDictionaryWithComparer.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Collections.Specialized
+{
+    // This class is an internal class used by System.Diagnostics.Proccess on property EnvironmentVariables which returns an StringDictionary. Since we need an OrdinalIgnoreCase
+    // comparer for non casesensitive environments and Ordinal comparer for case sensitive environments, we need this workaround in order to use StringDictionary. We should 
+    // override StringDictionary's method and use our own Dictionary<string,string> with comparer since this class is internal and StringDictionary _contents Hashtable 
+    // is private and we don't want to make that field protected.
+
+    [Serializable]
+    internal sealed class StringDictionaryWithComparer : StringDictionary
```

Comment:
Nit: I wonder if the name of this class/file should change as it's not really a `StringDictionaryWithComparer` anymore, it's more of a `StringDictionaryDictionaryWrapper` or maybe just `StringDictionaryWrapper`? Any better names?